shared_model/interfaces/iroha_internal/transaction_batch_template_definitions.hpp

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_TRANSACTION_BATCH_TEMPLATE_DEFINITIONS_HPP
+#define IROHA_TRANSACTION_BATCH_TEMPLATE_DEFINITIONS_HPP
+
+#include "interfaces/iroha_internal/transaction_batch.hpp"
+
+namespace shared_model {
+  namespace interface {
+
+    namespace {
+      /**
+       * check if all transactions belong to the same batch
+       * @param txs transactions to be checked
+       * @return true if all transactions from the same batch and false
+       * otherwise
+       */
+      bool allTxsInSameBatch(const types::SharedTxsCollectionType &txs) {
+        if (txs.size() == 1) {
+          return true;
+        }
+
+        // take batch meta of the first transaction and compare it with batch
+        // metas of remaining transactions
+        auto batch_meta = txs.front()->batchMeta();
+        if (not batch_meta) {
+          return false;
+        }
+
+        return std::none_of(++txs.begin(),
+                            txs.end(),
+                            [front_batch_meta = batch_meta.value()](
+                                const std::shared_ptr<Transaction> tx) {
+                              return tx->batchMeta()
+                                  ? **tx->batchMeta() != *front_batch_meta
+                                  : false;
+                            });
+      };
+    }  // namespace
+
+    template <typename TransactionValidator, typename FieldValidator>
+    iroha::expected::Result<TransactionBatch, std::string>
+    TransactionBatch::createTransactionBatch(
```

Comment:
Add documentation for the function

Reply:
Documentation is in interface file, `transaction_batch.hpp`